packages/@lwc/template-compiler/src/codegen/helpers.ts

```diff
@@ -6,19 +6,113 @@
  */
 import * as t from '../shared/estree';
 import { toPropertyName } from '../shared/utils';
-import { BaseElement, ChildNode, LWCDirectiveRenderMode, Node } from '../shared/types';
+import {
+    Attribute,
+    BaseElement,
+    ChildNode,
+    Element,
+    LWCDirectiveRenderMode,
+    Node,
+    Property,
+} from '../shared/types';
 import {
     isParentNode,
     isSlot,
     isForBlock,
     isBaseElement,
     isIf,
     isDynamicDirective,
+    isElement,
+    isText,
+    isComment,
 } from '../shared/ast';
 import { TEMPLATE_FUNCTION_NAME, TEMPLATE_PARAMS } from '../shared/constants';
 
 import CodeGen from './codegen';
 
+function serializeAttrs(element: Element): string {
+    /**
+     * 0: styleToken in existing class attr
+     * 1: styleToken for added class attr
+     * 2: styleToken as attr
+     */
+    const attrs: string[] = [];
+    let hasClassAttr = false;
+
+    const collector = ({ name, value }: Attribute | Property) => {
+        let v = (value as any).value;
+
+        if (name === 'class') {
+            hasClassAttr = true;
+            v += '${0}';
+        }
+        if (typeof v === 'string') {
+            attrs.push(`${name}="${v}"`);
```

Comment:
This also needs to be escaped. 🙂 

```html
<template>
    <h1 data-foo="&quot;>\x3Cscript>alert('pwned')\x3C/script>\x3Ch1 foo=&quot;">
    </h1>
</template>
```

![Screen Shot 2022-04-12 at 1 49 33 PM](https://user-images.githubusercontent.com/283842/163051971-dbd3dbe3-8bc6-4e1f-be64-b32fa0c11e27.png)